### DISCUSSION TITLE
memory frequency

### DISCUSSION THREAD

**Discussion by SudoDios** [oshi/oshi#2269] [Q&A]
Hi.
How to get memory frequency?
`clockSpeed` return -1

**Comment by dbwiddis** [comment]
The `clockSpeed` field on `PhysicalMemory` is OSHI's best attempt to get memory frequency.  If it's returning -1, OSHI was unable to determine it.

What operating system are you using?  How can you determine the frequency without OSHI?

**Reply by SudoDios** [reply]
Thank you.
I using Ubuntu 20.04.
And it can be `sudo lshw -short -C memory`

**Comment by dbwiddis** [comment]
OSHI is fetching it with `dmidecode --type 17`, but both that and the `lshw` approach you cite require `sudo` to work properly.  Do you know of a way to get the value without elevated permissions?

**Reply by SudoDios** [reply]
So what is the problem with sudo?

**Reply by SudoDios** [reply]
Can't solve the problem with jni with a C source?

**Reply by dbwiddis** [reply]
If you run OSHI under sudo you will get the frequency result.

The main point is the information is not available without elevated privileges.

**Comment by dbwiddis** [comment]
Looks like [it is not possible](https://unix.stackexchange.com/questions/608397/get-memory-type-and-speed-without-sudo)
